src/errors/publicErrors.ts

```diff
@@ -54,6 +54,17 @@ export class EntityNotFoundError extends PublicNonRecoverableError {
   }
 }
 
+export class EntityGoneError extends PublicNonRecoverableError {
```

Comment:
do we ever need to instanceof check it? wonder if we need a flag on it, and a typeguard

Reply:
Good point, yeah, we would need the `instanceof` in some cases

Reply:
typeguard added, not sure what do you mean by the flag, but please have a look to see if it makes sense 🙏 

Note: all 410 http errors considered as entityGone, no matter if it was created as a PublincNonRecoverableError

Reply:
common technique for simple typeguards is adding a field like "isEntityGone: true" so that you could do a single-field check, but you are right, in this case http code check is a better option!